### PR TITLE
Drop support for Python <=3.8 and TensorFlow <=2.4

### PR DESCRIPTION
**Pull request by lgeiger**
This drops support for version 3.8 and older which also means that we now require TensorFlow 2.5 and newer as it is the first release that shipped with wheels for Python 3.9.

See https://github.com/larq/compute-engine/pull/786#issuecomment-1668196582

### CODE REVIEW

File: larq_compute_engine/tflite/python/interpreter_base.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
 import numpy as np
 from tqdm import tqdm
 
-Data = Union[np.ndarray, List[np.ndarray]]
+Data = Union[np.ndarray, list[np.ndarray]]
 
 
-def data_generator(x: Union[Data, Iterator[Data]]) -> Iterator[List[np.ndarray]]:
+def data_generator(x: Union[Data, Iterator[Data]]) -> Iterator[list[np.ndarray]]:
```
Comment (Tombana) [inline larq_compute_engine/tflite/python/interpreter_base.py:10]: Reminder to ourselves that we should also upstream these changes to our internal repo.

**Review by Tombana — APPROVED** [review]
Nice!

**Comment by lgeiger** [comment]
I started a test release [here](https://github.com/larq/compute-engine/actions/runs/5795274252) let's see what CI thinks about this.

**Review by CNugteren — APPROVED** [review]

**Comment by lgeiger** [comment]
> I started a test release [here](https://github.com/larq/compute-engine/actions/runs/5795274252) let's see what CI thinks about this.

Looks like the build was successful apart from some issues with the Android benchmarker build: #788

**Review by Tombana — APPROVED** [review]

**Pull request merged**